### PR TITLE
Syntax highlighting

### PR DESCRIPTION
**Pull request by sjagoe**
Naive syntax highlighting.

Due to the original pager design, so highlighting runs on every draw. As a result, it has two main issues:
1. ~A lot slower than it needs to be~
2. ~Syntax highlighting can break (show errors) when scrolling down a buffer~

This now progressively processes the input for highlighting, but on a large file, if `End` is pushed, then it blocks to consume the whole input. But now my CPU usage is quite high while idle (5%), as it no longer blocks on user input and "ticks" every 50ms

### CODE REVIEW

File: src/app/blob_pager.rs [hunk 1]
```diff
@@ -26,13 +48,42 @@ impl<'repo> BlobPager<'repo> {
...
+        let syntax_set = two_face::syntax::extra_newlines();
+        let theme_set = two_face::theme::extra();
+
+        let syntax = match Path::new(&name).extension().and_then(OsStr::to_str) {
+            Some(ext) => syntax_set.find_syntax_by_extension(ext).cloned(),
+            _ => match lines.first() {
+                Some(line) => syntax_set.find_syntax_by_first_line(line).cloned(),
+                _ => None,
+            },
+        };
```
Comment (jrasanen) [inline src/app/blob_pager.rs]: While concise, I find this visually hard to parse/read. Perhaps something like this would be easier to read? What do you think?

```suggestion
        let syntax = {
            let extension = Path::new(&name).extension().and_then(OsStr::to_str);
            if let Some(ext) = extension {
                syntax_set.find_syntax_by_extension(ext).cloned()
            } else if let Some(line) = lines.first() {
                syntax_set.find_syntax_by_first_line(line).cloned()
            } else {
                None
            }
        };
```

**Review by jrasanen — COMMENTED** [review]
Nice work! I left one suggestion.

Something to wonder about in the future is to support light colors for the syntax highlighting.

**Comment by sjagoe** [comment]
> support light colors for the syntax highlighting

I was thinking at some point I'll properly integrate syntect themes for the whole app. Right now I just used the theme from examples.
